src/error/mod.rs

```diff
@@ -0,0 +1,167 @@
+use std::error::Error;
+use std::fmt;
+
+pub type NetavarkResult<T> = Result<T, NetavarkError>;
+
+// The main Netavark error type
+#[derive(Debug)]
+pub enum NetavarkError {
+    // A string message
+    Message(String),
+    // A string message that sets a specific exit code for Netavark
+    ExitCode(String, i32),
+    // A chain of multiple errors
+    Chain(String, Box<NetavarkError>),
+
+    Io(std::io::Error),
+
+    Dbus(zbus::Error),
+    DbusVariant(zvariant::Error),
+
+    Sysctl(sysctl::SysctlError),
+
+    Serde(serde_json::Error),
+}
+
+// Internal struct for JSON output
+#[derive(Debug, Serialize, Deserialize)]
+struct JsonError {
+    pub error: String,
+}
+
+impl NetavarkError {
+    // TODO: There has to be a better way of doing this
+    pub fn from_str(string: &str) -> NetavarkError {
+        NetavarkError::Message(string.to_string())
+    }
+
+    // TODO: There has to be a better way of doing this
+    pub fn make_chain(string: String, chained: NetavarkError) -> NetavarkError {
```

Comment:
can we name this `wrap` instead to better match the go term?

Reply:
Sure